packages/atomic/src/components/search-box-suggestions/atomic-search-box-recent-queries/atomic-search-box-recent-queries.tsx

```diff
@@ -1,48 +1,171 @@
-              onClick: () => recentQueriesList.executeRecentQuery(i),
-            })),
-        };
-      }, this.host);
+      this.initialize();
     } catch (error) {
       this.error = error as Error;
     }
   }
 
+  private initialize() {
+    dispatchSearchBoxSuggestionsEvent((bindings) => {
+      this.bindings = bindings;
+      this.recentQueriesList = buildRecentQueriesList(bindings.engine, {
+        initialState: {queries: this.retrieveLocalStorage()},
+        options: {maxLength: 1000},
+      });
+
+      this.recentQueriesList.subscribe(() => this.updateLocalStorage());
+
+      return {
+        position: Array.from(this.host.parentNode!.children).indexOf(this.host),
+        onInput: () => {},
+        renderItems: () => this.renderItems(),
+      };
+    }, this.host);
+  }
+
+  private retrieveLocalStorage() {
+    return JSON.parse(
+      window.localStorage.getItem(localStorageKey) || '[]'
+    ) as string[];
+  }
+
+  private updateLocalStorage() {
+    if (!this.recentQueriesList.state.analyticsEnabled) {
+      return this.disableFeature();
+    }
+
+    window.localStorage.setItem(
+      localStorageKey,
+      JSON.stringify(this.recentQueriesList.state.queries)
+    );
+  }
+
+  private disableFeature() {
+    if (!this.userWarned) {
+      this.userWarned = true;
+      this.bindings.engine.logger.warn(
+        'Because analytics are disabled, the recent queries feature is deactivated.'
+      );
+    }
+
+    window.localStorage.removeItem(localStorageKey);
+  }
+
+  private renderItems(): SearchBoxSuggestionElement[] {
+    if (!this.recentQueriesList.state.analyticsEnabled) {
+      return [];
+    }
+
+    const query = this.bindings.searchBoxController.state.value;
+    const hasQuery = query !== '';
+    const max = hasQuery ? this.maxWithQuery : this.maxWithoutQuery;
+    const filteredQueries = this.recentQueriesList.state.queries
+      .filter(
+        (recentQuery) =>
+          recentQuery !== query &&
+          recentQuery.toLowerCase().startsWith(query.toLowerCase())
+      )
+      .slice(0, max);
+
+    const suggestionElements = filteredQueries.map((value) =>
+      this.renderItem(value)
+    );
+    if (suggestionElements.length) {
+      suggestionElements.unshift(this.renderClear());
+    }
+
+    return suggestionElements;
+  }
+
+  private renderClear(): SearchBoxSuggestionElement {
+    return {
+      key: 'recent-query-clear',
+      content: (
+        <div class="flex justify-between w-full">
+          <span class="font-bold">
+            {this.bindings.i18n.t('recent-searches')}
+          </span>
+          <span>{this.bindings.i18n.t('clear')}</span>
+        </div>
+      ),
+      onSelect: () => {
+        this.recentQueriesList.clear();
+        this.bindings.triggerSuggestions();
+      },
+    };
+  }
+
+  private renderItem(value: string): SearchBoxSuggestionElement {
+    const query = this.bindings.searchBoxController.state.value;
+    return {
+      key: value,
```

Comment:
just in case other suggestions may happen to have the same query, I think this should be prefixed or something else.

```suggestion
      key: 'recent-' + value,
```